e3nn/o3/tensor_product/_codegen.py

```diff
@@ -1,14 +1,37 @@
-from typing import List, Tuple
+import re
 from math import sqrt
-import textwrap
+from typing import List, Tuple
+
+import torch
+from torch import fx
 
 from e3nn import o3
-from e3nn.util.codegen import LazyCodeGenerator
 from e3nn.util import prod
 
 from ._instruction import Instruction
 
 
+def _get_code(graph):
+    x = graph.python_code('')
+    x = x.replace('def forward(self, ', '@torch.jit.script\ndef main(')
+    x = x.replace('Ellipsis', '...')
+    x = x.replace('slice(None, None, None)', ':')
+    x = re.sub(r'slice\(None, (-?\d+), None\)', r':\1', x)
+    x = re.sub(r'slice\((\d+), (-?\d+), None\)', r'\1:\2', x)
+    x = x.replace('[(', '[')
+    x = x.replace(')]', ']')
```

Comment:
```suggestion
```
`Ellipsis` is the only problem for torchscript, the other regexes can be removed

Reply:
indeed